db/migrate/20200324130825_change_date_to_timestamp.rb

```diff
@@ -0,0 +1,198 @@
+class ChangeDateToTimestamp < ActiveRecord::Migration
+  def up
+    safety_assured do
+      change_column :access_tokens, :created_at, :datetime
```

Comment:
At first, I thought you would do something more dynamic, for example like this:
```ruby
class ChangeDateToTimestamp < ActiveRecord::Migration[5.0]
  def up
    safety_assured do
      ApplicationRecord.connection.tables.each do |table_name|
        ApplicationRecord.connection.columns(table_name).each do |column|
          next if column.type != 'date' # or perhaps compare the column.limit here :)
          change_column table_name, column.name, :datetime
        end
      end
    end
  end
end
```

But anyway, I trust your list 👍

Reply:
I did it :) But I want to verify the list :)

```ruby
require './config/environment'

Rails.application.eager_load!

changes = ActiveRecord::Base.descendants.each_with_object({}) do |model, memo|
  next if model.abstract_class?

  columns = model.columns.select do |c|
    c.type == :datetime
  end

  memo[model.table_name] = columns.map(&:name)
end

changes = changes.sort

File.open('migration.rb', 'w+') do |f|
  changes.each do |table, names|
    names.each do |name|

      f.puts "change_column :#{table}, :#{name}, :datetime"
    end
  end
end
```

Reply:
You will miss some tables if you just use ApplicationRecord btw ;)

Reply:
Gold! 👏